Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp

```diff
@@ -349,15 +349,17 @@ void SourceBufferPrivateRemote::setAppendWindowEnd(const MediaTime& appendWindow
     gpuProcessConnection->connection().send(Messages::RemoteSourceBufferProxy::SetAppendWindowEnd(appendWindowEnd), m_remoteSourceBufferIdentifier);
 }
 
-void SourceBufferPrivateRemote::computeSeekTime(const WebCore::SeekTarget& target, CompletionHandler<void(const MediaTime&)>&& completionHandler)
+Ref<SourceBufferPrivate::ComputeSeekPromise> SourceBufferPrivateRemote::computeSeekTime(const WebCore::SeekTarget& target)
 {
     auto gpuProcessConnection = m_gpuProcessConnection.get();
-    if (!isGPURunning()) {
-        completionHandler(MediaTime::invalidTime());
-        return;
-    }
+    if (!isGPURunning())
+        return ComputeSeekPromise::createAndReject(-1);
 
-    gpuProcessConnection->connection().sendWithAsyncReply(Messages::RemoteSourceBufferProxy::ComputeSeekTime(target), WTFMove(completionHandler), m_remoteSourceBufferIdentifier);
+    return gpuProcessConnection->connection().sendWithPromisedReply(Messages::RemoteSourceBufferProxy::ComputeSeekTime(target), m_remoteSourceBufferIdentifier)->whenSettled(RunLoop::main(), [](auto&& result) {
```

Comment:
Ditto here. For a result that's an `optional<>`, it could create or reject based on whether the optional has a value.

Reply:
the result is an `Expected<T, U>` The IPC Promise wraps the type set in the messages.in file, and an IPCError as exception.

It forces you to deal with the error if there's one.